definitions/subworkflows/cnvkit_single_sample.cwl

```diff
@@ -63,7 +70,14 @@ steps:
             scatter_plot: scatter_plot
             drop_low_coverage: drop_low_coverage
             male_reference: male_reference
-            reference_cnn: reference_cnn
+            reference:
+                source: [reference_cnn, fasta_reference]
+                valueFrom: |
+                    ${
+                      var cnn = self[0];
+                      var fasta = self[1];
+                      return {'cnn_file': cnn, 'fasta_file': fasta};
```

Comment:
This does not match either of the types available for this input.

Reply:
The reference input of `cnvkit_batch.cwl` is defined as one of the following:
* A `record` containing a `cnn_file`.
* A `record` containing a `fasta_file` and an optional `normal_bam`.

Just like `fasta_file` oughn't to be given both a `string` and `File` simultaneously, the type definition is satisfied by either one of these two records.  The record with `cnn_file` has no `fasta_file` and the record with `fasta_file` has no `cnn_file`.

Reply:
I'm using the exclusive input parameter as described in https://www.commonwl.org/user_guide/11-records/index.html
if both records are provided it will take the first record type. In this case that's the record with the `cnn_file`

Reply:
After a quick test I concede that, although `cwltool` produces an error message in this situation (like:
```
echo12.yaml:3:5: invalid field `thing2`, expected one of: 'thing1'
```
), it seems to carry on regardless.

That being the case I might prefer to specifically give it one or the other parameter here while we're already scripting, but it seems to not be strictly necessary.

Reply:
Are you suggesting to add more javascript to check if the `cnn_file` is non null? 
if it is, pass just `cnn_file: cnn`
if it is not, pass just `fasta_file: fasta`

or are you suggesting basically 2 cnvkit workflows, 
1 for `reference.cnn` method
1 for the flat reference and tumor/normal methods 

Trying to make the workflows easy to use and configure but don't want to make it super complicated!

Reply:
I was imagining the "more javascript" route.  The complexity remains hidden inside the code that way.  If it works fine in Cromwell, though, maybe changing it isn't necessary at all.

Reply:
I'm hesitant to add more javascript. I'm bad at debugging javascript in the cwl files and the errors Cromwell displays are not useful. 
If you have a solution I can add it and rerun my tests.